libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -220,48 +300,95 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
-          const filtersParameters = lowerCaseFilters["parameters"];
+        if (featureFlagValue.conditions && featureFlagValue.conditions.clientFilters) {
+          for (let i = 0; i < featureFlagValue.conditions.clientFilters.length; i++) {
+            const parameters: Partial<ClientFilter> = {};
+            //
+            // Converting client_filter keys to lower case
+            const lowerCaseFilters = this.getLowerCaseFilters(
+              featureFlagValue.conditions.clientFilters[i]
+            );
 
-          if (filtersName) {
-            if (filtersName.toLowerCase() == "alwayson") {
-              featureFlagValue.conditions = defaultFeatureConditions;
-              break;
+            const filtersName = lowerCaseFilters["name"];
+            const filtersParameters = lowerCaseFilters["parameters"];
+
+            if (filtersName) {
+              if (filtersName.toLowerCase() == "alwayson") {
+                featureFlagValue.conditions = defaultFeatureConditions;
+                break;
+              }
+              parameters.name = filtersName;
+              if (filtersParameters) {
+                parameters.parameters = filtersParameters;
+              }
+
+              featureFlagValue.conditions.clientFilters[i] = parameters as ClientFilter;
             }
-            parameters.name = filtersName;
-            if (filtersParameters) {
-              parameters.parameters = filtersParameters;
+            else {
+              throw new ArgumentError(
+                `This feature flag '${featureFlagName}' has a filter without the required 'name' property.`
+              );
             }
-
-            featureFlagValue.conditions.clientFilters[i] = parameters as ClientFilter;
-          }
-          else {
-            throw new ArgumentError(
-              `This feature flag '${featureFlagName}' has a filter without the required 'name' property.`
-            );
           }
         }
       }
+
+      const requirementType = featureData[requirementTypeKeyWord];
+
+      if (requirementType) {
+        if (this.validateRequirementType(featureFlagName, requirementType)) {
+          featureFlagValue.conditions.requirementType = requirementType;
+        }
+      }
     }
-    else if (typeof featureData[featureFlagName] == "boolean") {
-      featureFlagValue.enabled = featureData[featureFlagName];
+    else if (typeof featureData == "boolean") {
+      featureFlagValue.enabled = featureData;
     }
     else {
       throw new ArgumentError(
         `Data contains feature flags in invalid format. The type of ${featureFlagName} should be either boolean or dictionary.`
       );
     }
+
     return featureFlagValue;
   }
 
+  private getFeatureFlagValueFromMsFmSchema(featureFlag: any): MsFeatureFlagValue {
+    const validate = this.ajv.compile<MsFeatureFlagValue>(MsFeatureFlagValueSchema);
+    const featureFlagCopy = JSON.parse(JSON.stringify(featureFlag)); //deep copy
```

Comment:
`const featureFlagCopy = JSON.parse(JSON.stringify(featureFlag)); //deep copy` am okay with what we currently have looks good to me, we can also use this[ _cloneDeep](https://lodash.com/docs/4.17.15#cloneDeep) function from lodash library